app/services/gift_service.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module GiftService
+  module_function
+
+  def call
+    user_dates = []
+
+    User.where(state: 'incompleted').find_in_batches do |group|
+      group.each do |u|
+        last_date = u.receipt.max_by do |receipt|
```

Comment:
The object returned here is not a date, it's a hash of a PR
```suggestion
        last_pr = u.receipt.max_by do |receipt|
```